core/src/main/java/org/elasticsearch/search/highlight/CustomQueryScorer.java

```diff
@@ -27,14 +27,23 @@
 import org.apache.lucene.spatial.geopoint.search.GeoPointInBBoxQuery;
 import org.elasticsearch.common.lucene.search.function.FiltersFunctionScoreQuery;
 import org.elasticsearch.common.lucene.search.function.FunctionScoreQuery;
-import org.elasticsearch.index.query.HasChildQueryBuilder;
 import org.elasticsearch.index.query.HasChildQueryParser;
 
 import java.io.IOException;
 import java.util.Map;
 
 public final class CustomQueryScorer extends QueryScorer {
 
+    private static final Class<?> unsupportedGeoQuery;
+
+    static {
+        try {
+            unsupportedGeoQuery = Class.forName("org.apache.lucene.spatial.geopoint.search.GeoPointMultiTermQuery");
```

Comment:
can you add a comment that this class in pkg private and that's why we are making this stunt?